### PR TITLE
simgrid 3.32

### PR DESCRIPTION
**Pull request by Porkepix**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

Updated url for a stable one. Also moved archive format to a `.tar.bz2` which seems better than the (still available) `.tar.gz` although I'm not sure what maintainers generally prefer here: I've got the same issue on other formulas, what's the order of preference between `.tar.gz`, `.tar.bz2`, `.tar.xz`, `.zip` and others?

### CODE REVIEW

**Comment by Porkepix** [comment]
@chenrui333 Why prefer the archive from PyPI rather than from the repository itself?

**Comment by chenrui333** [comment]
> @chenrui333 Why prefer the archive from PyPI rather than from the repository itself?

we can then use the builtin livecheck without querying against the homepage.

@samford @nandahkrishna What do you think?

**Comment by samford** [comment]
> > @chenrui333 Why prefer the archive from PyPI rather than from the repository itself?
> 
> we can then use the builtin livecheck without querying against the homepage.

Unless we have problems checking the existing `stable` source in livecheck (doesn't appear to be the case here), I'm generally not in favor of arbitrarily switching the `stable` source simply for the sake of livecheck. In this case, I prefer to keep the `stable` source aligned with what the first-part website links to and that isn't PyPI.

In this case, I would say that we should continue to use the tarball link from the homepage's "Download Source" link (i.e., https://framagit.org/simgrid/simgrid/uploads/c45f7fd6872b3b0d26b9ba2e607d6e3a/simgrid-3.32.tar.gz). The tarball from the repository release (used in the initial commit here) is missing a few files in comparison to the uploaded tarball linked from the homepage (though I can't say whether that has a practical effect without testing). Sure, we generally favor the file format with the smallest download size (and the bzip2 *is* ~500 KB smaller than the gzip file) but that's when we're choosing between files with identical contents.

**Comment by Porkepix** [comment]
> > > @chenrui333 Why prefer the archive from PyPI rather than from the repository itself?
> > 
> > 
> > we can then use the builtin livecheck without querying against the homepage.
> 
> Unless we have problems identifying versions from the existing `stable` source in livecheck (doesn't appear to be the case here), I'm generally not in favor of arbitrarily switching the `stable` source simply for the sake of livecheck. In this case, I prefer to keep the `stable` URL aligned with what the first-part website links to and that isn't PyPI.
> 
> In this case, I would say that we should continue to use the tarball link from the homepage's "Download Source" link (i.e., [framagit.org/simgrid/simgrid/uploads/c45f7fd6872b3b0d26b9ba2e607d6e3a/simgrid-3.32.tar.gz](https://framagit.org/simgrid/simgrid/uploads/c45f7fd6872b3b0d26b9ba2e607d6e3a/simgrid-3.32.tar.gz)). The tarball from the repository release (used in the initial commit here) is missing a few files in comparison to the uploaded tarball linked from the homepage (though I can't say whether that has a practical effect without testing). Sure, we generally favor the file format with the smallest download size (and the bzip2 _is_ ~500 KB smaller than the gzip file) but that's when we're choosing between files with identical contents.

I had this intermediate commit https://github.com/Homebrew/homebrew-core/pull/112340/commits/f609512187879575d48dd7c3b46f5962e388ddea using an asset link on the release rather than an upload (no idea why they do those uploads), see on https://framagit.org/simgrid/simgrid/-/releases/v3.32 links in the assets section opposed the uploads at the bottom. I've verified the checksums to be identical, but the assets url is stable and can be updated by the bump command itself while the upload one can't.

Don't know what's different in the tarball you're talking about though.

**Comment by samford** [comment]
> I had this intermediate commit [f609512](https://github.com/Homebrew/homebrew-core/commit/f609512187879575d48dd7c3b46f5962e388ddea) using an asset link on the release rather than an upload (no idea why they do those uploads), see on https://framagit.org/simgrid/simgrid/-/releases/v3.32 links in the assets section opposed the uploads at the bottom. I've verified the checksums to be identical, but the assets url is stable and can be updated by the bump command itself while the upload one can't.
> 
> Don't know what's different in the tarball you're talking about though.

I may not have been clear. I was comparing the contents of the "uploads" file (e.g., the `.tar.gz` file linked on the homepage and at the bottom of the release) to the automatically-generated release tarballs in the "Assets" section of the release. These files don't have identical contents but I had it backwards: the assets tarball has extra files that the uploads tarball omits.

I did builds using both and the differences in the resulting files are as follows (omitting expected binary differences between builds):

* `/include/simgrid/version.h` has a different value for `#define SIMGRID_GIT_VERSION` (on line 11). It's `"e8d2ff8dab"` in the uploads build and `"none, release version"` for the assets build.
* The assets build also includes a `COPYRIGHT.template` file in the installed files.

I don't use `simgrid` and I'm not familiar with the formula, so I'll defer to someone else regarding what's preferable here. Continuing to use the same tarball is the easy path here but, if these changes aren't meaningful, then there's also something to be said for switching to the assets file to be able to use `brew bump` with this formula.

If there's support for the latter, I can update the `livecheck` block to check releases from their GitLab repository (from `releases.json`).

-----

That said, the long-term goal is for `livecheck` to be able to capture the full URL from the match and provide this to `brew bump` but that will require a good amount of work. Outside of the internal changes, we would need to extend existing regexes to match the rest of the URL (e.g., `\.t` would need to change to something like `\.t[^"' >]+?) and livecheck would need to be able to effectively isolate the URL from the matched text without an additional capture group (as we expect the first capture group to be the version). This has been on my very long to-do list for a while but there are a number of things ahead of it (and I'm currently tied up fixing the broken checks from the GitHub release assets HTML change anyway).

**Comment by Porkepix** [comment]
> > I had this intermediate commit [f609512](https://github.com/Homebrew/homebrew-core/commit/f609512187879575d48dd7c3b46f5962e388ddea) using an asset link on the release rather than an upload (no idea why they do those uploads), see on [framagit.org/simgrid/simgrid/-/releases/v3.32](https://framagit.org/simgrid/simgrid/-/releases/v3.32) links in the assets section opposed the uploads at the bottom. I've verified the checksums to be identical, but the assets url is stable and can be updated by the bump command itself while the upload one can't.
> > Don't know what's different in the tarball you're talking about though.
> 
> I may not have been clear. I was comparing the contents of the "uploads" file (e.g., the `.tar.gz` file linked on the homepage and at the bottom of the release) to the automatically-generated release tarballs in the "Assets" section of the release. These files don't have identical contents but I had it backwards: the assets tarball has extra files that the uploads tarball omits.

Actually that's completely my bad here, I thought I compared both archives and saw them as having the same checksum, but it seems I fucked up somewhere: I couldn't verify this here when I rechecked.

I also checked and it seems PyPI have a third different archive: checksum is both different from the upload and the assets…
So I have no idea here if current url should be kept, or the one I suggested, or PyPI's as suggested by @chenrui333

**Comment by samford** [comment]
> I also checked and it seems PyPI have a third different archive: checksum is both different from the upload and the assets…

The contents of the PyPI tarball are the same as the uploads tarball with the addition of a few insignificant files, so we can basically treat those as the same (they should produce similar output). From my perspective, this is still between the uploads tarball and the assets tarball (i.e., I still don't think we should use the PyPI URL simply for the sake of livecheck in this instance).

**Comment by carlocab** [comment]
+1 for the uploads tarball, since that's what upstream packaged for distribution.

For archive formats: `.gz` is more portable so we should probably stick to those unless either a) the size difference is significant (i.e. in the order of tens of MBs), or b) the formula already has the necessary dependency available at build time (i.e. `xz` for `.xz`, `bzip2` for `.bz2`, etc.).

Note that (b) is really only an issue on Linux (macOS `tar` can extract `.tar.{xz,bz2}` archives), so if the formula doesn't have many installs on Linux (which is typically the case for most formulae) then we can probably ignore this consideration too.

**Comment by Porkepix** [comment]
> +1 for the uploads tarball, since that's what upstream packaged for distribution.

Even if that means `brew bump` can't update it? Of the three archives that's the only one `brew bump` can't act on.

For the archives question, is it still true with the updated Ubuntu CI to 22.04? I believe I moved some xorg formulas to `.tar.xz` that had no dependency on `xz` without any issue. The problem is still existing on several libs using `tar.bz2` that's not distributed anymore, now they distribute `.tar.gz` or `.tar.xz` only.

**Comment by chenrui333** [comment]
Cool, based on the discussions in here, I am gonna revert pypi source change.

**Review by chenrui333 — DISMISSED** [review]

File: Formula/simgrid.rb [hunk 1]
```diff
@@ -3,9 +3,9 @@ class Simgrid < Formula
 
   desc "Studies behavior of large-scale distributed systems"
   homepage "https://simgrid.org/"
-  url "https://framagit.org/simgrid/simgrid/uploads/caf09286c8e698d977f11e8f8451ba46/simgrid-3.31.tar.gz"
-  sha256 "4b44f77ad40c01cf4e3013957c9cbe39f33dec9304ff0c9c3d9056372ed4c61d"
-  revision 3
+  url "https://framagit.org/simgrid/simgrid/-/archive/v3.32/simgrid-v3.32.tar.bz2"
+  sha256 "b544b33ce13324c4c50bb76ce912db8f2fd874100ecda5e55a34e5cbc62641ce"
```
Comment (samford) [inline Formula/simgrid.rb:7]:
```suggestion
  url "https://framagit.org/simgrid/simgrid/uploads/c45f7fd6872b3b0d26b9ba2e607d6e3a/simgrid-3.32.tar.gz"
  sha256 "837764eb81562f04e49dd20fbd8518d9eb1f94df00a4e4555e7ec7fa8aa341f0"
```

**Comment by samford** [comment]
I'm not very knowledgeable about licenses but, looking at the [`COPYING` file](https://framagit.org/simgrid/simgrid/-/blob/master/COPYING), this may involve more than just `LGPL-2.1-only` (i.e., I'm not sure if that license supercedes the others).

**Comment by Porkepix** [comment]
So, who can tell for the license question?
Also, regarding the comment about `fails_with gcc: "5"`?

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**